lib/bitstyles_phoenix/bitstyles.ex

```diff
@@ -25,11 +27,24 @@ defmodule BitstylesPhoenix.Bitstyles do
   # If it does exist, then doing this renaming in the classname/2 function would make it impossible
   # for users of older bitstyles to use the "class-name-A" classname.
 
-  def classname(class, version) when version >= "5.0.0" do
+  defp downgrade_classname(class, target_version, _current_version)
+       when target_version > {5, 0, 1} do
+    IO.warn("Version #{version_to_string(target_version)} of bitstyles is not yet supported")
+    class
+  end
+
+  defp downgrade_classname(class, target_version, _current_version)
```

Comment:
I am a bit confused by this, since I thought this would be as simple as:
```
def downgrade_classname(class, version, version), do: class

def downgrade_classname(class, {5, 0, 0}, target_version) do
  # map to 4,2,0
  classnames(class, {4, 2, 0}, target_version)
end

def downgrade_classname(class, {4, 2, 0}, target_version) do
  # map to 4,1,0
  classnames(class, {4, 1, 0}, target_version)
end

#...
```
After all, all releases are known and we do not need to support anything in between.

Or am I missing something?

Reply:
We just should then not skip a released version :D

Reply:
> We just should then not skip a released version :D

This would mean adding a lot of do-nothing implementations of `downgrade_classname`, e.g. for downgrades from:
6.1.0 to 6.0.0
5.0.1 to 5.0.0
4.3.0 to 4.2.0
4.1.0 to 4.0.0
4.0.0 to 3.1.0
3.1.0 to 3.0.0

that doesn't sound very appealing to me 😅

Reply:
Maybe not appealing but definitely way less complex. And for these thing I would value simplicity over most other things. After all a do-nothing implementation would always document in the code that nothing is happening for that version, so it even adds to the readability imho.

Reply:
Done: [`24de682` (#134)](https://github.com/bitcrowd/bitstyles_phoenix/pull/134/commits/24de6821b47d205f8e0a918fe16c97b42fd4e1f6)